astropy/stats/bayesian_blocks.py

```diff
@@ -45,19 +45,36 @@
    https://www.tandfonline.com/doi/abs/10.1080/01621459.1969.10501038
 """
 
+from __future__ import annotations
+
 import warnings
 from inspect import signature
+from typing import TYPE_CHECKING
 
 import numpy as np
 
 from astropy.utils.exceptions import AstropyUserWarning
 
+if TYPE_CHECKING:
+    from collections.abc import KeysView
+    from typing import Literal
+
+    from numpy.typing import ArrayLike, NDArray
+
+# TODO: typing: use a custom-defined 'ArrayLike-but-not-a-scalar' type for `float | ArrayLike` or `ArrayLike | float` hints
+
 # TODO: implement other fitness functions from appendix C of Scargle 2013
 
 __all__ = ["FitnessFunc", "Events", "RegularEvents", "PointMeasures", "bayesian_blocks"]
 
 
-def bayesian_blocks(t, x=None, sigma=None, fitness="events", **kwargs):
+def bayesian_blocks(
+    t: ArrayLike,
+    x: ArrayLike | None = None,
+    sigma: ArrayLike | float | None = None,
+    fitness: Literal["events", "regular_events", "measures"] | FitnessFunc = "events",
+    **kwargs,
+) -> NDArray[float]:
```

Comment:
I think this might be my fault, but unfortunately `NDArray[float]` is not technically correct as `float` is not a `numpy.dtype`. See https://numpy.org/doc/stable/reference/typing.html#numpy.typing.NDArray, but this actually needs to be 
``NDArray[np.floating]`` (or however `np.floating` is imported).

Reply:
No problem, I should've double-checked. Fixed throughout `stats` in [e9425d9](https://github.com/astropy/astropy/pull/16562/commits/e9425d9244a662f1dee5f6cfc7a53b30b4a291b0)